Packaging for openSUSE: 7.1.0 with build errors on i586 and aarch64

Dear maintainers,

I updated the openSUSE package for the opencloud-server to 7.1.0 and noticed, that there are build errors on i586 and aarch64 (see logs below).

Both seem to be from the `Rollup` utility, which (to me as a non-nodejs expert) looks like it might just not be available for those architectures.

I have no idea if this was just introduced in 7.1.0. And I am lacking the knowledge to switch the build to use the WASM mode suggested by the error message.

What do you think?

Kind Regards,
Johannes

## i586

```bash
[   46s] + pnpm build
[   47s] 
[   47s] > identifier@1.0.0 build /home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp
[   47s] > node --openssl-legacy-provider scripts/build.js && rm -f build/service-worker.js
[   47s] 
[   47s] ◇ injected env (4) from .env // tip: ⌘ multiple files { path: ['.env.local', '.env'] }
[   48s] /home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/native.js:140
[   48s] 	throw new Error(
[   48s] 	^
[   48s] 
[   48s] Error: Your current platform "linux" and architecture "ia32" combination is not yet supported by the native Rollup build. Please use the WASM build "@rollup/wasm-node" instead.
[   48s] 
[   48s] The following platform-architecture combinations are supported:
[   48s] android-arm
[   48s] android-arm64
[   48s] darwin-arm64
[   48s] darwin-x64
[   48s] freebsd-arm64
[   48s] freebsd-x64
[   48s] linux-arm
[   48s] linux-arm (musl)
[   48s] linux-arm64
[   48s] linux-arm64 (musl)
[   48s] linux-loong64
[   48s] linux-loong64 (musl)
[   48s] linux-ppc64
[   48s] linux-ppc64 (musl)
[   48s] linux-riscv64
[   48s] linux-riscv64 (musl)
[   48s] linux-s390x
[   48s] linux-x64
[   48s] linux-x64 (musl)
[   48s] openbsd-x64
[   48s] openharmony-arm64
[   48s] win32-arm64
[   48s] win32-ia32
[   48s] win32-x64
[   48s] 
[   48s] If this is important to you, please consider supporting Rollup to make a native build for your platform and architecture available.
[   48s]     at throwUnsupportedError (/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/native.js:140:8)
[   48s]     at getPackageBase (/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/native.js:131:3)
[   48s]     at Object.<anonymous> (/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/native.js:93:21)
[   48s]     at Module._compile (node:internal/modules/cjs/loader:1812:14)
[   48s]     at Object..js (node:internal/modules/cjs/loader:1943:10)
[   48s]     at Module.load (node:internal/modules/cjs/loader:1533:32)
[   48s]     at Module._load (node:internal/modules/cjs/loader:1335:12)
[   48s]     at wrapModuleLoad (node:internal/modules/cjs/loader:255:19)
[   48s]     at Module.require (node:internal/modules/cjs/loader:1556:12)
[   48s]     at require (node:internal/modules/helpers:152:16)
[   48s] 
[   48s] Node.js v24.14.1
[   48s]  ELIFECYCLE  Command failed with exit code 1.
[   48s] error: Bad exit status from /var/tmp/rpm-tmp.XbDMgM (%build)
```

### aarch64

```bash
[   69s] + pnpm build
[   70s] 
[   70s] > identifier@1.0.0 build /home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp
[   70s] > node --openssl-legacy-provider scripts/build.js && rm -f build/service-worker.js
[   70s] 
[   70s] ◇ injected env (4) from .env // tip: ⌁ auth for agents [www.vestauth.com]
[   71s] /home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/native.js:115
[   71s] 		throw new Error(
[   71s] 		^
[   71s] 
[   71s] Error: Cannot find module @rollup/rollup-linux-arm64-gnu. npm has a bug related to optional dependencies (https://github.com/npm/cli/issues/4828). Please try `npm i` again after removing both package-lock.json and node_modules directory.
[   71s]     at requireWithFriendlyError (/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/native.js:115:9)
[   71s]     at Object.<anonymous> (/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/native.js:124:76)
[   71s]     at Module._compile (node:internal/modules/cjs/loader:1706:14)
[   71s]     at Object..js (node:internal/modules/cjs/loader:1839:10)
[   71s]     at Module.load (node:internal/modules/cjs/loader:1441:32)
[   71s]     at Function._load (node:internal/modules/cjs/loader:1263:12)
[   71s]     at TracingChannel.traceSync (node:diagnostics_channel:328:14)
[   71s]     ... 2 lines matching cause stack trace ...
[   71s]     at require (node:internal/modules/helpers:147:16) {
[   71s]   [cause]: Error: Cannot find module '@rollup/rollup-linux-arm64-gnu'
[   71s]   Require stack:
[   71s]   - /home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/native.js
[   71s]   - /home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/shared/parseAst.js
[   71s]   - /home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/shared/rollup.js
[   71s]   - /home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/rollup.js
[   71s]   - /home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/workbox-build@7.4.1/node_modules/workbox-build/build/lib/bundle.js
[   71s]   - /home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/workbox-webpack-plugin@7.4.1_webpack@5.105.2/node_modules/workbox-webpack-plugin/build/generate-sw.js
[   71s]   - /home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/workbox-webpack-plugin@7.4.1_webpack@5.105.2/node_modules/workbox-webpack-plugin/build/index.js
[   71s]   - /home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/ui_config/webpack.config.js
[   71s]   - /home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/scripts/build.js
[   71s]       at Function._resolveFilename (node:internal/modules/cjs/loader:1383:15)
[   71s]       at defaultResolveImpl (node:internal/modules/cjs/loader:1025:19)
[   71s]       at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1030:22)
[   71s]       at Function._load (node:internal/modules/cjs/loader:1192:37)
[   71s]       at TracingChannel.traceSync (node:diagnostics_channel:328:14)
[   71s]       at wrapModuleLoad (node:internal/modules/cjs/loader:237:24)
[   71s]       at Module.require (node:internal/modules/cjs/loader:1463:12)
[   71s]       at require (node:internal/modules/helpers:147:16)
[   71s]       at requireWithFriendlyError (/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/native.js:97:10)
[   71s]       at Object.<anonymous> (/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/native.js:124:76) {
[   71s]     code: 'MODULE_NOT_FOUND',
[   71s]     requireStack: [
[   71s]       '/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/native.js',
[   71s]       '/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/shared/parseAst.js',
[   71s]       '/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/shared/rollup.js',
[   71s]       '/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/rollup@4.60.4/node_modules/rollup/dist/rollup.js',
[   71s]       '/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/workbox-build@7.4.1/node_modules/workbox-build/build/lib/bundle.js',
[   71s]       '/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/workbox-webpack-plugin@7.4.1_webpack@5.105.2/node_modules/workbox-webpack-plugin/build/generate-sw.js',
[   71s]       '/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/node_modules/.pnpm/workbox-webpack-plugin@7.4.1_webpack@5.105.2/node_modules/workbox-webpack-plugin/build/index.js',
[   71s]       '/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/ui_config/webpack.config.js',
[   71s]       '/home/abuild/rpmbuild/BUILD/opencloud-server-7.1.0-build/opencloud-server-7.1.0/services/idp/scripts/build.js'
[   71s]     ]
[   71s]   }
[   71s] }
[   71s] 
[   71s] Node.js v22.22.0
[   71s]  ELIFECYCLE  Command failed with exit code 1.
```

--- a/services/idp/pnpm-workspace.yaml
+++ b/services/idp/pnpm-workspace.yaml
@@ -3,6 +3,22 @@ autoInstallPeers: true
 publicHoistPattern:
   - "*"
 
+# Ensure platform-specific optional dependencies (e.g. Rollup's native
+# binaries pulled in via workbox) are fetched for every architecture we
+# build/package for, not just the host that runs `pnpm install`. Without
+# this, cross-arch RPM/OBS builds fail with "Cannot find module
+# @rollup/rollup-linux-arm64-gnu" on aarch64.
+supportedArchitectures:
+  os:
+    - current
+    - linux
+  cpu:
+    - x64
+    - arm64
+  libc:
+    - glibc
+    - musl
+
 overrides:
   fast-uri: ">=3.1.2"
   postcss: ">=8.5.10"

